### PR TITLE
feat: more events tracking for SaaS

### PR DESCRIPTION
**Pull request by scmmishra**
## Description

This PR adds more event to track across the dashboard

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested locally with a June account for development

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/helper/AnalyticsHelper/events.js [hunk 1]
```diff
@@ -1,9 +1,69 @@
...
+export const CONVERSATION_EVENTS = {
+  EXECUTED_A_MACRO: 'Executed a macro',
+  SENT_MESSAGE: 'Sent a message',
+  SENT_PRIVATE_NOTE: 'Sent a private note',
+  INSERTED_A_CANNED_RESPONSE: 'Inserted a canned response',
+  USED_MENTIONS: 'Used mentions',
+  MERGED_CONTACTS: 'Used merge contact option',
+  ADDED_TO_CANNED_RESPONSE: 'Used added to canned response option',
+  ADDED_A_CUSTOM_ATTRIBUTE: 'Added a custom attribute',
+  ADDED_AN_INBOX: 'Added an inbox',
```
Comment (pranavrajs) [inline app/javascript/dashboard/helper/AnalyticsHelper/events.js:11]: Not a conversation action.

File: app/javascript/dashboard/helper/AnalyticsHelper/events.js [hunk 2]
```diff
@@ -1,9 +1,69 @@
...
+  CREATE_CONTACT: 'Create a contact',
+  IMPORT_MODAL_OPEN: 'Open import contacts modal',
+  IMPORT_FAILURE: 'Import contacts failed',
+  IMPORT_SUCCESS: 'Import contacts success',
+};
+
+// CAMPAIGN EVENTS
+export const CAMPAIGNS_EVENTS = {
+  OPEN_NEW_CAMPAIGN_MODAL: 'Open new campaign modal',
+  CREATE_CAMPAIGN: 'Create new campaign',
```
Comment (pranavrajs) [inline app/javascript/dashboard/helper/AnalyticsHelper/events.js:42]:
```suggestion
  CREATE_CAMPAIGN: 'Created a new campaign',
```

File: app/javascript/dashboard/helper/AnalyticsHelper/events.js [hunk 3]
```diff
@@ -1,9 +1,69 @@
...
+  CREATE_CATEGORY: 'Create a new category',
+  DELETE_CATEGORY: 'Delete a category',
+  EDIT_CATEGORY: 'Edit a category',
+
+  CREATE_ARTICLE: 'Create a new article',
+  PUBLISH_ARTICLE: 'Publish an article',
+  UPDATE_ARTICLE_SETTINGS: 'Update article settings',
+  ARCHIVE_ARTICLE: 'Archive an article',
+  DELETE_ARTICLE: 'Delete an article',
+  PREVIEW_ARTICLE: 'Preview article',
```
Comment (pranavrajs) [inline app/javascript/dashboard/helper/AnalyticsHelper/events.js:68]: Let's use in it past tense, following the same pattern as before.

File: app/javascript/dashboard/helper/AnalyticsHelper/events.js (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/helper/AnalyticsHelper/events.js:9]: Move `MERGED_CONTACTS` to the `CONTACTS_EVENTS` category in https://github.com/chatwoot/chatwoot/pull/6234/commits/346707bb2084eb71950e1cc24212abf52e70842b

About the rest, should I create a separate category called `CORE_EVENTS` or `ACCOUNT_EVENTS`?

**Review by fayazara — APPROVED** [review]
Tested locally with a staging account created on june, tracked page navigations, actions etc.

Working as expected.

File: app/javascript/dashboard/helper/AnalyticsHelper/events.js (hunk 1 above)
Comment (pranavrajs) [reply app/javascript/dashboard/helper/AnalyticsHelper/events.js:9]: Let us go with ACCOUNT_EVENTS

Comment (scmmishra) [reply app/javascript/dashboard/helper/AnalyticsHelper/events.js:12]: Done

**Comment by scmmishra** [comment]
@pranavrajs need a final review on this and we can roll it out

**Comment by scmmishra** [comment]
@pranavrajs @fayazara the codeclimate failure is on the places where the tracker is implemented, the tracking utilities have 100% coverage. can we go ahead and merge it?

**Pull request merged**